common/include/nephtys/utils/config.hpp

```diff
@@ -0,0 +1,45 @@
+//
+// Created by sztergbaum roman on 2019-04-19.
+//
+
+#pragma once
+
+#include <filesystem>
+#include <fstream>
+#include <nlohmann/json.hpp>
+#include <loguru.hpp>
+
+namespace nephtys::utils
+{
+    template<typename TConfig>
+    TConfig load_configuration(std::filesystem::path &&config_path, std::string filename) noexcept
+    {
+        VLOG_SCOPE_F(loguru::Verbosity_INFO, __FUNCTION__);
+        nlohmann::json config_json_data;
+        TConfig loaded_config{};
+        const auto &fullpath = config_path / std::move(filename);
+        DVLOG_F(loguru::Verbosity_INFO, "path to nephtys configuration -> %s", fullpath.string().c_str());
+        if (!std::filesystem::exists(config_path)) {
+            DVLOG_F(loguru::Verbosity_WARNING,
```

Comment:
What about splitting the config creation / config loading into two different function ?

Reply:
@doom

Reply:
It seems okay to me as it is now